src/api/app/controllers/webui/cloud/upload_jobs_controller.rb

```diff
@@ -0,0 +1,72 @@
+module Webui
+  module Cloud
+    class UploadJobsController < WebuiController
+      before_action :require_login
+      before_action -> { feature_active?(:cloud_upload) }
+      before_action :set_user
+      before_action :validate_configuration_presence
+      before_action :set_breadcrump
+
+      def index
+        @upload_jobs = @user.upload_jobs
+      end
+
+      def new
+        @project_name = params[:project]
+        @package_name = params[:package]
+        @repository_name = params[:repository]
+        @architecture_name = params[:architecture]
+        @file_name = params[:filename]
```

Comment:
You could drop this and access the params from the view instead.

And should we validate the parameters before using them in the view? :thinking:

Reply:
I would prefer to not hand the params to the view!

We can do some validation / sanity check in the future, for this simple prototype I didn't do it on purpose. But yes, in the future we should check for the existence of the parameter (not sure if this is actually possible with the file).

Reply:
> ChrisBr 20 hours ago 
> I would prefer to not hand the params to the view!

@ChrisBr But why? It's the same thing, except that you temporarly store them in a variable.

Reply:
> @ChrisBr But why? It's the same thing, except that you temporarly store them in a variable.

Bjoern has a point there..

But I also think that even better would be to try and mimic a standard ActiveRecord controller. So have a new method which instantiates the uploadjob model i.e.
```ruby
def new
  @upload_job = ::Cloud::UploadJob.new(@user, {})
end
```

And then in create have it function just like create methods usually do with activerecord models i.e. rendering the new template if the upload job is invalid:
```ruby
def create
  @upload_job = ::Cloud::UploadJob.new(@user, params).create

  if @upload_job.valid?
    flash[:success] = "Successfully created upload job #{@upload_job.id}."
    redirect_to cloud_upload_index_path
  else
    flash[:error] = "Failed to create upload job: #{@upload_job.errors.full_messages.to_sentence}."
    render :new
  end
end
```

Reply:
And then in the view you can do this:
```ruby
= form_for @upload_job, url: cloud_upload_index_path, method: :post do
  = f.hidden_field :project
```

Reply:
> @ChrisBr But why? It's the same thing, except that you temporarly store them in a variable.
Because it is a mix of concerns, the params hash does not belong to the view!

Reply:
I will add @evanrolfe suggestions next year, however, there are a few problems:

- We need to use Cloud::Backend::UploadJob as Cloud::UploadJob does not know about the attributes
- Cloud::Backend::UploadJob can only be created with an xml at the moment

So this needs some adaptions which I won't have time to implement before X-Mas ...